src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,225 @@
+  // Use hardcoded gas estimate values
+  // Values set by looking at a couple of different ERC-20 deposits
+  bn(240_000)
+
+export async function getTradeQuote(
+  input: GetEvmTradeQuoteInput,
+): Promise<Result<TradeQuote, SwapErrorRight>> {
+  const {
+    chainId,
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    supportsEIP1559,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+  } = input
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
+    throw new Error(`Arbitrum Bridge only supports EVM chains`)
+  }
+
+  const adapter = assertGetEvmChainAdapter(chainId)
+
+  const l2Network = await getL2Network(arbitrum.id)
+
+  const isDeposit = sellAsset.chainId === ethChainId
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+  const isTokenBridge = !isEthBridge
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+  const erc20L1Address = fromAssetId((isDeposit ? sellAsset : buyAsset).assetId).assetReference
+  const erc20L2Address = fromAssetId((isDeposit ? buyAsset : sellAsset).assetId).assetReference
+  const l1Provider = getEthersV5Provider(KnownChainIds.EthereumMainnet)
+  const l2Provider = getEthersV5Provider(KnownChainIds.ArbitrumMainnet)
+
+  if (isTokenBridge) {
+    // Since our related assets list isn't exhaustive and won't cut it to determine the L1 <-> L2 mapping, we double check that the bridge is valid
+    // by checking against Arbitrum bridge's own mappings, which uses different sources (Coingecko, Gemini, Uni and its own lists at the time of writing)
+    const arbitrumBridgeErc20L2Address = await (bridger as Erc20Bridger).getL2ERC20Address(
+      erc20L1Address,
+      l1Provider,
+    )
+    const arbitrumBridgeErc20L1Address = await (bridger as Erc20Bridger).getL1ERC20Address(
+      erc20L2Address,
+      l2Provider,
+    )
+
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L1Address), getAddress(erc20L1Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L1 ERC20 address: ${erc20L1Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L2Address), getAddress(erc20L2Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L2 ERC20 address: ${erc20L2Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+  }
+
+  if (sellAsset.assetId === usdcAssetId || sellAsset.assetId === usdcOnArbitrumAssetId) {
+    // https://www.circle.com/en/cross-chain-transfer-protocol
+    throw new Error('cctp not implemented')
+  }
+
+  if (isDeposit && !isEthBridge) {
```

Comment:
we can maybe improve this a bit with an allowance check and using the fallback gas only if allowance is insufficient and leverage the sdk otherwise.

Reply:
Was absolutely thinking about that, but not a fan of adding more logic in this guy which already does a lot - from my testing, since like fallback fees consistently overestimate, which isn't a big deal atm as we're talking e.g 30 bucks instead of 25.

Will try and add it in the cleanest way possible, as I agree, it would be better to actually estimate the Tx when possible

Reply:
https://github.com/shapeshift/web/pull/6977/commits/ae57872de7371812330b00aebdddcf300aeb4f94  - note this isn't doing allowance checks per se, but letting simulation fail and using fallback then